### PR TITLE
Tweaks to getCommands

### PR DESCRIPTION
**Pull request by grabbou**
Removes dependencies on any real files and mocks everything.

### CODE REVIEW

File: test/getCommands.spec.js [hunk 1]
```diff
@@ -2,38 +2,53 @@ const path = require('path');
...
+  it('should export unique list of commands by name', () => {
+    mock(pjsonPath, {
+      dependencies: {
+        'rnpm-plugin-test': '*',
+        'rnpm-plugin-test-2': '*',
+      },
+    });
+    mock(testPluginPath, commands.single);
+    mock(
+      path.join(process.cwd(), 'node_modules', 'rnpm-plugin-test-2'),
```
Comment (Kureev) [inline test/getCommands.spec.js]: What requires `rnpm-plugin-test-2`?

File: test/getCommands.spec.js [hunk 2]
```diff
@@ -2,38 +2,53 @@ const path = require('path');
...
+      commands.single
+    );
+    expect(getCommands().length).to.be.equal(1);
+  });
+
+  afterEach(mock.stopAll);
```
Comment (Kureev) [inline test/getCommands.spec.js]: I'd keep before/after hooks together. It's like keeping at the same place `componentWillMount` and `componentWillReceiveProps`

File: test/getCommands.spec.js (hunk 1 above)
Comment (grabbou) [inline test/getCommands.spec.js]: Nothing, we want to define two plugins that export the same thing `commands.single` to check if we return an unique list (case when a plugin defines `rnpm link` but there's already one)

File: test/getCommands.spec.js (hunk 2 above)
Comment (grabbou) [inline test/getCommands.spec.js]: But 99% of our test suite works like that already, so it's more a question if it's a part of this PR or another one.

Comment (Kureev) [inline test/getCommands.spec.js]: K, can be scoped to another one

**Comment by grabbou** [comment]
so mock or mockRequire ?

File: test/getCommands.spec.js (hunk 1 above)
Comment (grabbou) [inline test/getCommands.spec.js]: https://github.com/rnpm/rnpm/blob/feature/support-multiple-actions/src/getCommands.js#L12

**Comment by Kureev** [comment]
well, here we don't use a `mockFs`, so we can use `mock-require` without suffix

**Comment by grabbou** [comment]
Ok, so this is fixed now.

**Pull request merged**